### PR TITLE
Convert models to macro templates

### PR DESCRIPTION
**Pull request by dbirman**
This PR overhauls the aind-data-schema-models repository to use `jinja2` templates. 

The entry point is the `run_all.sh` script, which pulls the templates from the `_generators/templates` folder. The name of each template file is used to get the corresponding data from the `models` folder. The templates then parse each row, generating the corresponding classes, and merge them into the ALL/ONE_OF patterns that we use. It's not super easy to write the templates so I almost certainly made a typo somewhere, especially in the docstrings appreciate a close eye on them. 

Some additional utility functions were added to support this and tests to coverage the new code.

Overall I'm happy with this approach, it's super simple and the templates are pretty understandable except for some weird edge cases with how empty columns are treated by pandas. This should be backward compatible with our existing repos, but we need to check carefully before merging this.

Also the `mouse_anatomy` file is massive, although it's only really a few MB so not a big deal to keep it. I vote that we merge this in some form and then revisit how to handle large vocabularies with this issue https://github.com/AllenNeuralDynamics/aind-data-schema-models/issues/80 in the future.

### CODE REVIEW

**Comment by dbirman** [comment]
Closes #73

**Comment by dbirman** [comment]
Closes #63

File: src/aind_data_schema_models/_generators/generator.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    Parameters
+    ----------
+    data_type : str
+        Which template file to use
+    isort : bool, optional
+        Whether to run isort on the output, by default True
+    black : bool, optional
+        Whether to run black on the output, by default True
+    """
+    ROOT_DIR = Path("./src/aind_data_schema_models/")
```
Comment (jtyoung84) [inline src/aind_data_schema_models/_generators/generator.py:21]: Can we make the input path configurable as an input arg?

File: tests/test_generator.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        # Define the paths that will be used in the function
+        data_type = "test_data"
+        output_path = Path("./src/aind_data_schema_models/test_data.py")
+
+        # Run the function with isort and black enabled
+        generate_code(data_type, isort=True, black=True)
+
+        # Check if the CSV file was read correctly
+        mock_read_csv.assert_called_once_with(Path(f"./src/aind_data_schema_models/_generators/models/{data_type}.csv"))
```
Comment (jtyoung84) [inline tests/test_generator.py:34]: Sometimes it's convenient to run the tests individually. You can set an absolute path to this file like it's done here:
https://github.com/AllenNeuralDynamics/aind-data-schema-models/blob/main/tests/test_utils.py
which makes it easy to run the tests from different directories and different operating systems.

File: src/aind_data_schema_models/_generators/generator.py (hunk 1 above)
Comment (dbirman) [reply src/aind_data_schema_models/_generators/generator.py]: added

File: tests/test_generator.py (hunk 2 above)
Comment (dbirman) [reply tests/test_generator.py]: added

**Comment by jtyoung84** [comment]
Seems good to me. I want to import this branch into a few repos and run the unit tests in those to see if everything passes.

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
The pid_names module seems to have been removed in this PR.

**Comment by jtyoung84** [comment]
The sub-lists in the Organizations are missing. Stuff like DETECTOR_MANUFACTURERS, SUBJECT_SOURCES, etc.: https://github.com/AllenNeuralDynamics/aind-data-schema-models/blob/main/src/aind_data_schema_models/organizations.py

**Comment by dbirman** [comment]
> The sub-lists in the Organizations are missing. Stuff like DETECTOR_MANUFACTURERS, SUBJECT_SOURCES, etc.: https://github.com/AllenNeuralDynamics/aind-data-schema-models/blob/main/src/aind_data_schema_models/organizations.py

Whoops thanks for catching that, I'll add a test too so we don't lose them again...

**Review by jtyoung84 — APPROVED** [review]
I think it should be good to go. There was one small issue where a warning message was displaying
`Union[_Callithrix_Jacchus, _Homo_Sapiens,...]`
instead of
`Union[CALLITHRIX_JACCHUS, HOMO_SAPIENS,...]`
but I think that should be okay.

**Comment by dbirman** [comment]
> I think it should be good to go. There was one small issue where a warning message was displaying `Union[_Callithrix_Jacchus, _Homo_Sapiens,...]` instead of `Union[CALLITHRIX_JACCHUS, HOMO_SAPIENS,...]` but I think that should be okay.

I think I changed the class names by accident, but the dynamically generated classes probably shouldn't have been named in all caps anyways...
